test/functional/tests/io_class/test_io_class_stats_file_size_core.py

```diff
@@ -0,0 +1,201 @@
+                      "and load it for all caches."):
+        io_classes = prepare_io_classes(caches)
+
+    with TestRun.step("Run fio for all devices checking IO class statistics."):
+        sizes = get_sizes(io_classes)
+        size_min = Size(512, Unit.Byte)
+
+        io_class_direct = [io_class for io_class in io_classes if "direct" in io_class.rule][0]
+
+        for io_class, size in zip(io_classes[2:-1], sizes):
+            with TestRun.step(f"Testing IO class {io_class.id}."):
+                with TestRun.step("Run fio for each device with direct=1 and check that "
+                                  "for direct IO class statistics increases."):
+                    for core in cores:
+                        core.reset_counters()
+                        fio = fio_params(core, size_min, size, direct=True)
+                        result = fio.run()
+                        sync()
+                        drop_caches()
+                        fio_stats = \
+                            result[0].write_requests_number() + result[0].read_requests_number()
+                        io_class_stats = core.get_io_class_statistics(io_class_direct.id).\
+                            request_stats.requests_total
+                        check_statistics(core, io_class_direct, fio_stats, io_class_stats)
+                        fs_utils.remove(f"{core.mount_point}/*", force=True, recursive=True)
+                    size_min = size + Unit.Byte
+
+
+def fio_params(core, size_min, size_max, direct=False):
+    fio = Fio().create_command() \
+        .io_engine(IoEngine.libaio) \
+        .do_verify() \
+        .time_based() \
+        .run_time(timedelta(seconds=64) if size_max <= Unit.GibiByte else timedelta(seconds=200)) \
+        .verification_with_pattern() \
+        .read_write(ReadWrite.randrw) \
+        .write_percentage(100) \
+        .io_depth(16) \
+        .block_size(Size(1, Unit.Blocks512)) \
+        .direct(direct)
+    fio.add_job() \
+        .file_size(size_min) \
+        .target(
+        f"{core.mount_point}/{round(size_min.get_value())}{'_direct' if direct else ''}")
+    fio.add_job() \
+        .file_size(size_max) \
+        .target(
+        f"{core.mount_point}/{round(size_max.get_value())}{'_direct' if direct else ''}")
+
+    return fio
+
+
+def check_statistics(core, io_class, fio_stats, io_class_stats):
+    TestRun.LOGGER.info(f"Checking IO class stats for cache {core.cache_id}.")
+    if io_class_stats < fio_stats:
+        TestRun.LOGGER.error(f"Total requests too low for IO Class {io_class.id} "
+                             f"{io_class.rule}: {io_class_stats}, fio requests: {fio_stats}")
+    else:
+        TestRun.LOGGER.info(f"Total requests for IO Class {io_class.id} "
+                            f"{io_class.rule}: {io_class_stats}, fio requests: {fio_stats}")
+
+
+def prepare_devices():
+    cache_device = TestRun.disks['cache']
+    cache_device.create_partitions([Size(10, Unit.GibiByte)] * 5)
```

Comment:
I suggest limiting the number of caches to the number of filesystem the tests are using.

Reply:
yup, the test could be parametrized by CacheMode and use one cache and multiple cores (one per tested filesystem)